src/staticList.js

```diff
@@ -0,0 +1,190 @@
+  console.log(`Fetching ${url}`);
+  const res = await fetch(url);
+
+  if (!res.ok) {
+    throw new Error(`Error fetching list ${url}`, res);
+  }
+
+  const json = await res.json();
+  return json;
+}
+
+async function writeFile(dir, filename, input) {
+  const data = JSON.stringify(input, null, 4);
+  const filePath = path.join(dir, filename);
+
+  console.log(`Writing file ${filePath}`);
+
+  if (!fs.existsSync(dir)) {
+    fs.mkdirSync(dir, { recursive: true });
+  }
+
+  fs.writeFileSync(filePath, data);
+}
+
+function sleep(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function getTokenLists() {
+  // Get token lists
+  const cowList = await fetchFromCoingecko(cowListUrl);
+  const coinGeckoList = await fetchFromCoingecko(coinGeckoListUrl);
+
+  // Combine 2 lists
+  const combined = [...cowList.tokens, ...coinGeckoList.tokens];
+
+  console.log("Combined tokens", combined.length);
+
+  return combined;
+}
+
+async function getIds() {
+  // If ID's file exist
+  if (fs.existsSync(IDS_FILE_PATH)) {
+    const json = fs.readFileSync(IDS_FILE_PATH, "utf8");
+    return JSON.parse(json);
+  }
+
+  const combined = await getTokenLists();
+
+  // Get Coingecko id list
+  const coinGeckoIdList = await fetchFromCoingecko(coinGeckoIdListUrl);
+
+  // Crate unique list by symbols from those 2
+  const uniq = _.uniqBy(combined, ({ symbol }) => symbol);
+  // Get just an array of symbols
+  const symbols = uniq.map(({ symbol }) => symbol.toLowerCase());
+
+  // Get coin gecko list of ids for that uniq list
+  const idsData = coinGeckoIdList.filter((el) =>
+    symbols.includes(el.symbol.toLowerCase())
+  );
+
+  // Write ID's file
+  writeFile(LIST_DIR, IDS_FILE_NAME, idsData);
+
+  return idsData;
+}
+
+async function getStaticData(idsData) {
+  // If static data file exists, read and return it
+  if (fs.existsSync(STATIC_LIST_PATH)) {
+    const json = fs.readFileSync(STATIC_LIST_PATH, "utf8");
+    return JSON.parse(json);
+  }
+
+  const successDelay = 3000; // Delay in milliseconds for successful requests
+  const errorDelay = 10000; // Delay in milliseconds for error requests
+  const staticData = [];
+
+  let index = 0;
+  while (index < idsData.length) {
+    const idItem = idsData[index];
+    const quoteParams = `localization=false&tickers=false&market_data=true&community_data=false&developer_data=false&sparkline=false`;
+    const url = `https://api.coingecko.com/api/v3/coins/${idItem.id}?${quoteParams}`;
+
+    console.log(`Fetching data for ID: ${idItem.id}`);
+
+    try {
+      const data = await fetchFromCoingecko(url);
+      staticData.push(data);
+      await sleep(successDelay); // Delay after a successful fetch
+      index++; // Move to the next item
+    } catch (error) {
+      console.log(error);
+      console.error(`Error fetching data for ID: ${idItem.id}`);
+      console.log(`Retrying fetch for ID: ${idItem.id}`);
+      await sleep(errorDelay); // Delay after an error fetch
+    }
+  }
```

Comment:
Idea to improve this.

Since we expect a large quantity of requests, we could throttle from the start (or only after a rate limit).
If using lodash already: https://docs-lodash.com/v4/throttle/